### PR TITLE
Migrate exceptions to common4j

### PR DESCRIPTION
**Pull request by rpdome**
Migrate exceptions to common4j

ADAL: https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/1618
MSAL: https://github.com/AzureAD/microsoft-authentication-library-for-android/pull/1451
Broker: https://github.com/AzureAD/ad-accounts-for-android/pull/1610

### CODE REVIEW

File: changelog.txt [hunk 1]
```diff
@@ -1,5 +1,6 @@
 V.Next
 ----------
+- [MAJOR] Migrate Exceptions (#1442)
```
Comment (shahzaibj) [inline changelog.txt:3]:
```suggestion
- [MAJOR] Migrate Exceptions from common to common4j (#1442)
```

File: common/src/main/java/com/microsoft/identity/common/adal/internal/util/HashMapExtensions.java [hunk 2]
```diff
@@ -134,26 +134,6 @@ private HashMapExtensions() {
...
-        if (webResponse != null && !TextUtils.isEmpty(webResponse.getBody())) {
-            JSONObject jsonObject = new JSONObject(webResponse.getBody());
-            Iterator<?> i = jsonObject.keys();
-            while (i.hasNext()) {
-                String key = (String) i.next();
-                response.put(key, jsonObject.getString(key));
-            }
-        }
-        return response;
-    }
```
Comment (shahzaibj) [inline common/src/main/java/com/microsoft/identity/common/adal/internal/util/HashMapExtensions.java:155]: no longer needed?

Comment (rpdome) [reply common/src/main/java/com/microsoft/identity/common/adal/internal/util/HashMapExtensions.java:155]: Moved to HashMapExtensions in Common4j.

**Review by AdamBJohnsonx — APPROVED** [review]

File: common4j/src/main/com/microsoft/identity/common/java/util/HashMapExtensions.java [hunk 3]
```diff
@@ -20,21 +20,36 @@
...
+public class HashMapExtensions {
 
-    @Builder.Default
-    String errorCode = "ERROR_CODE";
-
-    @Override
-    public String getErrorCode() {
-        return errorCode;
+    /**
+     * get key value pairs from response.
```
Comment (shahzaibj) [inline common4j/src/main/com/microsoft/identity/common/java/util/HashMapExtensions.java:36]:
```suggestion
     * Get key value pairs from response.
```

File: common4j/src/main/com/microsoft/identity/common/java/util/HashMapExtensions.java [hunk 4]
```diff
@@ -20,21 +20,36 @@
...
+     *
+     * @param webResponse {@link HttpResponse} to convert to a map
+     * @return HashMap
+     * @throws JSONException
+     */
+    public static HashMap<String, String> getJsonResponse(HttpResponse webResponse) throws JSONException {
+        final HashMap<String, String> response = new HashMap<>();
+        if (webResponse != null && !StringUtil.isNullOrEmpty(webResponse.getBody())) {
+            JSONObject jsonObject = new JSONObject(webResponse.getBody());
```
Comment (shahzaibj) [inline common4j/src/main/com/microsoft/identity/common/java/util/HashMapExtensions.java:45]:
```suggestion
            final JSONObject jsonObject = new JSONObject(webResponse.getBody());
```

File: common4j/src/main/com/microsoft/identity/common/java/util/HashMapExtensions.java [hunk 5]
```diff
@@ -20,21 +20,36 @@
...
+            Iterator<?> i = jsonObject.keys();
```
Comment (shahzaibj) [inline common4j/src/main/com/microsoft/identity/common/java/util/HashMapExtensions.java:46]:
```suggestion
            final Iterator<?> i = jsonObject.keys();
```

**Review by shahzaibj — APPROVED** [review]

**Pull request merged**